entitlement/src/main/java/org/killbill/billing/entitlement/api/DefaultEntitlement.java

```diff
@@ -715,19 +705,10 @@ public Entitlement changePlanWithDate(final EntitlementSpecifier spec, @Nullable
         // Get the latest state from disk
         refresh(callContext);
 
-        final BaseEntitlementWithAddOnsSpecifier baseEntitlementWithAddOnsSpecifier = new DefaultBaseEntitlementWithAddOnsSpecifier(
-                getBundleId(),
-                getBundleExternalKey(),
-                null,
-                effectiveDate,
-                effectiveDate,
-                false);
-        final List<BaseEntitlementWithAddOnsSpecifier> baseEntitlementWithAddOnsSpecifierList = new ArrayList<BaseEntitlementWithAddOnsSpecifier>();
-        baseEntitlementWithAddOnsSpecifierList.add(baseEntitlementWithAddOnsSpecifier);
         final EntitlementContext pluginContext = new DefaultEntitlementContext(OperationType.CHANGE_PLAN,
                                                                                getAccountId(),
                                                                                null,
-                                                                               baseEntitlementWithAddOnsSpecifierList,
+                                                                               null, //TODO_1375 - modified to pass null instead of baseEntitlementWithAddOnsSpecifierList as this would require changing baseEntitlementWithAddOnsSpecifier to return DateTime which is not required for now. Revisit later
```

Comment:
I see... that's ok, however, when you tackle the 'create' call, I suspect you will need to modify such object. I suppose this can be fixed at the same time.